ts/features/pn/store/reducers/payments.ts

```diff
@@ -0,0 +1,91 @@
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import { getType } from "typesafe-actions";
+import { Action } from "../../../../store/actions/types";
+import { UIMessageId } from "../../../../store/reducers/entities/messages/types";
+import { GlobalState } from "../../../../store/reducers/types";
+import {
+  remoteError,
+  remoteLoading,
+  remoteReady,
+  remoteUndefined,
+  RemoteValue
+} from "../../../bonus/bpd/model/RemoteValue";
+import { updatePaymentForMessage } from "../actions";
+import { Detail_v2Enum } from "../../../../../definitions/backend/PaymentProblemJson";
+import { PaymentRequestsGetResponse } from "../../../../../definitions/backend/PaymentRequestsGetResponse";
+import { RptId } from "../../../../../definitions/backend/RptId";
+
+export type MultiplePaymentState = {
+  [key: UIMessageId]: SinglePaymentState | undefined;
+};
+
+export type SinglePaymentState = {
+  [key: RptId]:
+    | RemoteValue<PaymentRequestsGetResponse, Detail_v2Enum>
+    | undefined;
+};
+
+export const initialState: MultiplePaymentState = {};
+
+export const paymentsReducer = (
+  state: MultiplePaymentState = initialState,
+  action: Action
+): MultiplePaymentState => {
+  switch (action.type) {
+    case getType(updatePaymentForMessage.request):
+      return {
+        ...state,
+        [action.payload.messageId]: {
+          ...state[action.payload.messageId],
+          [action.payload.paymentId]: remoteLoading
+        }
+      };
+    case getType(updatePaymentForMessage.success):
+      return {
+        ...state,
+        [action.payload.messageId]: {
+          ...state[action.payload.messageId],
+          [action.payload.paymentId]: remoteReady(action.payload.paymentData)
+        }
+      };
+    case getType(updatePaymentForMessage.failure):
+      return {
+        ...state,
+        [action.payload.messageId]: {
+          ...state[action.payload.messageId],
+          [action.payload.paymentId]: remoteError(action.payload.details)
+        }
+      };
+    case getType(updatePaymentForMessage.cancel):
+      return action.payload.reduce<MultiplePaymentState>(
+        (previousState, queuedUpdateActionPayload) => ({
+          ...previousState,
+          [queuedUpdateActionPayload.messageId]: {
+            ...previousState[queuedUpdateActionPayload.messageId],
+            [queuedUpdateActionPayload.paymentId]: undefined
```

Comment:
I think that **setting** to `remoteUndefined` is useful only if we want to keep track of payments that started an update and got cancelled later on, before they could be processed. This is because a payment for which an update was not requested will have a value of `undefined` (or, better said, will **not** have any value at all). At this point, I do not see any usefulness in tracking such cancelled-case.
By treating both cases with `undefined`, we are just saving memory since we are effectively removing data from the in-memory redux's state.